src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -8535,6 +8536,72 @@ class StubGenerator: public StubCodeGenerator {
+      //     <OP>     is the operation name, e.g. sin
+      //     <T>      is optional to indicate float/double
+      //              "f/d" for vector float/double operation
+      //     <N>      is the number of elements in the vector
+      //              "2/4" for neon, and "x" for sve
+      //     <U>      is the precision level
+      //              "u10/u05" represents 1.0/0.5 ULP error bounds
+      //               We use "u10" for all operations by default
+      //               But for those functions do not have u10 support, we use "u05" instead
+      //     <suffix> indicates neon/sve
+      //              "sve/advsimd" for sve/neon implementations
+      //     e.g. sinfx_u10sve is the method for computing vector float sin using SVE instructions
+      //          cosd2_u10advsimd is the method for computing 2 elements vector double cos using NEON instructions
+      //
+      log_info(library)("Loaded library %s, handle " INTPTR_FORMAT, JNI_LIB_PREFIX "vectormath" JNI_LIB_SUFFIX, p2i(libvectormath));
+
+      // Math vector stubs implemented with SVE for scalable vector size.
+      if (UseSVE > 0) {
+        for (int op = 0; op < VectorSupport::NUM_VECTOR_OP_MATH; op++) {
+          int vop = VectorSupport::VECTOR_OP_MATH_START + op;
+
+          // The native library does not support u10 level of "hypot".
+          const char* ulf = (vop == VectorSupport::VECTOR_OP_HYPOT) ? "u05" : "u10";
+
+          snprintf(ebuf, sizeof(ebuf), "%sfx_%ssve", VectorSupport::mathname[op], ulf);
+          StubRoutines::_vector_f_math[VectorSupport::VEC_SIZE_SCALABLE][op] = (address)os::dll_lookup(libvectormath, ebuf);
+
+          snprintf(ebuf, sizeof(ebuf), "%sdx_%ssve", VectorSupport::mathname[op], ulf);
+          StubRoutines::_vector_d_math[VectorSupport::VEC_SIZE_SCALABLE][op] = (address)os::dll_lookup(libvectormath, ebuf);
+        }
+      }
+
+      // Math vector stubs implemented with NEON for 64/128 bits vector size.
+      for (int op = 0; op < VectorSupport::NUM_VECTOR_OP_MATH; op++) {
+        int vop = VectorSupport::VECTOR_OP_MATH_START + op;
+
+        // The native library does not support u10 level of "hypot".
+        const char* ulf = (vop == VectorSupport::VECTOR_OP_HYPOT) ? "u05" : "u10";
+
+        snprintf(ebuf, sizeof(ebuf), "%sf4_%sadvsimd", VectorSupport::mathname[op], ulf);
+        StubRoutines::_vector_f_math[VectorSupport::VEC_SIZE_64][op] = (address)os::dll_lookup(libvectormath, ebuf);
+
+        snprintf(ebuf, sizeof(ebuf), "%sf4_%sadvsimd", VectorSupport::mathname[op], ulf);
+        StubRoutines::_vector_f_math[VectorSupport::VEC_SIZE_128][op] = (address)os::dll_lookup(libvectormath, ebuf);
+
+        snprintf(ebuf, sizeof(ebuf), "%sd2_%sadvsimd", VectorSupport::mathname[op], ulf);
+        StubRoutines::_vector_d_math[VectorSupport::VEC_SIZE_128][op] = (address)os::dll_lookup(libvectormath, ebuf);
+      }
+    } else {
+      log_info(library)("Failed to load native vector math library!");
```

Comment:
Include the `ebuf` message? The corresponding x86_64 code could also use a log message for the error case.